lib/puppet/reports/tagmail.rb

```diff
@@ -177,12 +173,25 @@ def process(tagmail_conf_file = "#{Puppet[:confdir]}/tagmail.conf")
       return
     end
 
-    metrics = raw_summary || {} rescue {}
-    metrics['resources'] = metrics['resources'] || {} rescue {}
-    metrics['events'] = metrics['events'] || {} rescue {}
-
-    if metrics['resources']['out_of_sync'] == 0 && metrics['resources']['changed'] == 0 && metrics['events']['audit'] == nil
-      Puppet.notice "Not sending tagmail report; no changes"
+    metrics = begin
+                raw_summary || {}
+              rescue
+                {}
+              end
+    metrics['resources'] = begin
+                             metrics['resources'] || {}
+                           rescue
+                             {}
+                           end
+    metrics['events'] = begin
+                          metrics['events'] || {}
+                        rescue
+                          {}
+                        end
+
+    if metrics['resources']['out_of_sync'] == 0 && metrics['resources']['changed'] == 0 && metrics['events']['audit'].nil? # rubocop:disable Style/NumericPredicate
```

Comment:
does metrics['resources']['out_of_sync'].zero? work ?

Reply:
Nope, causes a 'nil:NilClass' on the other two comparisons.

Reply:
other comparison****

Reply:
What I mean is that it causes 'nil:NilClass' errors on some comparisons.

Reply:
ah, because `nil.zero?` doesn't work but `nil == 0` would.